### ISSUE TITLE
Unexpected error when processing a restart file

### ISSUE THREAD

**Issue by MartinDix** [ACCESS-NRI/um2nc-standalone#32]
Trying to convert a field from a restart file fails
```
python  um2nc-standalone/src/um2netcdf.py --include 216 -- access-esm1.5-historical/archive/restart001/atmosphere/restart_dump.astart veg1.nc
...
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-24.01/lib/python3.10/site-packages/iris/cube.py", line 1515, in _check_multi_dim_metadata
    raise iris.exceptions.CannotAddError(msg)
iris.exceptions.CannotAddError: Invalid data dimensions: 2 given, 3 expected for 'surface_altitude'.
```
Same error when trying to convert the whole file.

Works ok with a CM2 file.

This problem is caused by presence of multiple surface orography fields in the restart file. The STASH processing adds them because it's saved to the monthly and daily mean output files. Iris doesn't handle this weird edge case properly.

**Comment by MartinDix** [comment]
After subsetting the file to just the prognostic variables get a different error

```
  File "/g/data/hh5/public/apps/miniconda3/envs/analysis3-24.01/lib/python3.10/site-packages/iris/fileformats/pp.py", line 791, in _data_bytes_to_shaped_array
    raise ValueError(emsg)
ValueError: PP field data containing 10865 words does not match expected length of 27840 words.
```

**Comment by truth-quark** [comment]
@MartinDix do you need the fix for https://github.com/ACCESS-NRI/UM_v7/issues/10 incorporated into `um2nc-standalone`?

**Comment by MartinDix** [comment]
> @MartinDix do you need the fix for [ACCESS-NRI/UM_v7#10](https://github.com/ACCESS-NRI/UM_v7/issues/10) incorporated into `um2nc-standalone`?

No, it's fix to the restart file in the model configuration. There's no practical workaround in um2nc.
